core/common/src/test/TileMetadata.test.ts

```diff
@@ -266,4 +268,126 @@ describe("TileMetadata", () => {
     test("4_0", "-c-5", { version: 4, noPatterns: true, externalTextures: true });
     test("a_1", "-F-2", { version: 10, projectExtents: true, noPatterns: true, externalTextures: true, instancing: true, elision: true });
   });
+
+  it("parseTileTreeIdAndContentId round trips", () => {
+    function test(modelId: Id64String, treeId: IModelTileTreeId, contentId: ReturnType<typeof ContentIdProvider.prototype.specFromId>, tileOptions: TileOptions) {
+      const treeIdStr = iModelTileTreeIdToString(modelId, treeId, tileOptions);
+      const contentIdStr = ContentIdProvider.create(true, tileOptions).idFromSpec(contentId);
+      const parsed = parseTileTreeIdAndContentId(treeIdStr, contentIdStr);
+
+      expect(parsed.modelId).to.equal(modelId);
+      expect(parsed.options).to.deep.equal(tileOptions);
+
+      // Sometimes ContentIdSpec and IModelTileTreeId will be slightly different due to "undefined" properties being included. However, this has no effect on further processing.
+      // Strings are compared here because the parsed values will typically be used to generate them.
+      expect(iModelTileTreeIdToString(parsed.modelId, parsed.treeId, parsed.options)).to.equal(treeIdStr);
+      expect(ContentIdProvider.create(true, parsed.options).idFromSpec(parsed.contentId)).to.equal(contentIdStr);
+    }
+
+    test("0x1c", { type: BatchType.Primary, edgesRequired: false } as PrimaryTileTreeId, { depth: 2, i: 5, j: 400, k: 16, multiplier: 8 }, defaultTileOptions);

```

Comment:
Curious - did compiler require you to cast to PrimaryTileTreeId and ClassifierTileTreeId here? Seems like it shouldn't.

Reply:
it didn't, but I found Intellisense to work better initially this way.